modular_skyrat/modules/ashwalkers/code/buildings/ash_tendril.dm

```diff
@@ -0,0 +1,139 @@
+		else
+			meat_counter++
+		playsound(get_turf(src),'sound/magic/demon_consume.ogg', 100, TRUE)
+		var/delivery_key = viewable_living.fingerprintslast //key of whoever brought the body
+		var/mob/living/delivery_mob = get_mob_by_key(delivery_key) //mob of said key
+		//there is a 40% chance that the Lava Lizard unlocks their respawn with each sacrifice
+		if(delivery_mob && (delivery_mob.mind?.has_antag_datum(/datum/antagonist/ashwalker)) && (delivery_key in ashies.players_spawned) && (prob(40)))
+			to_chat(delivery_mob, span_warning("<b>The Necropolis is pleased with your sacrifice. You feel confident your existence after death is secure.</b>"))
+			ashies.players_spawned -= delivery_key
+		viewable_living.gib()
+		atom_integrity = min(atom_integrity + max_integrity * 0.05, max_integrity) //restores 5% hp of tendril
+		for(var/mob/living/living_observers in view(src, 5))
+			if(living_observers.mind?.has_antag_datum(/datum/antagonist/ashwalker))
+				living_observers.add_mood_event("oogabooga", /datum/mood_event/sacrifice_good)
+			else
+				living_observers.add_mood_event("oogabooga", /datum/mood_event/sacrifice_bad)
+
+/**
+ * Proc that will spawn the egg that will revive the ashwalker
+ * This is also the Skyrat replacement for /proc/remake_walker
+ */
+/obj/structure/lavaland/ash_walker/proc/revive_ashwalker(mob/living/carbon/human/revived_ashwalker)
+	//spawn the egg we will move the body into
+	var/obj/structure/reviving_ashwalker_egg/spawned_egg = new(get_step(loc, pick(GLOB.alldirs)))
+	//move the body into the egg
+	revived_ashwalker.forceMove(spawned_egg)
+	to_chat(revived_ashwalker, span_warning("The tendril has decided to be merciful and revive you within a minute, have patience."))
+
+/obj/structure/reviving_ashwalker_egg
+	name = "occupied ashwalker egg"
+	desc = "Past the typical appearance of the yellow, man-sized egg, there seems to be a body floating within!"
+	icon = 'icons/mob/simple/lavaland/lavaland_monsters.dmi'
+	icon_state = "large_egg"
+	resistance_flags = LAVA_PROOF | FIRE_PROOF | FREEZE_PROOF
+	max_integrity = 80
+
+/obj/structure/reviving_ashwalker_egg/Initialize(mapload)
+	. = ..()
+	addtimer(CALLBACK(src, .proc/do_revive), 30 SECONDS)
+
+/**
+ * Proc that will fully revive the living content inside and then destroy itself
+ */
+/obj/structure/reviving_ashwalker_egg/proc/do_revive()
+	//try to find the living thing inside
+	var/mob/living/locate_living = locate() in contents
+	//if we can't locate a living thing, just delete the egg, its useless
+	if(!locate_living)
+		qdel(src)
+		return
+	//fully revive, "brand-spanking new" ashwalker
+	locate_living.revive(full_heal = TRUE, admin_revive = TRUE)
+	//force move the body to the outside
+	locate_living.forceMove(get_turf(src))
+	//force the ghost to re-enter the body
+	locate_living.mind.grab_ghost()
+	//give a nice alert to the viewers
+	locate_living.balloon_alert_to_viewers("[locate_living] breaks out of [src]!")
+	//goodbye egg
+	qdel(src)
```

Comment:
I don't think we need step by step comments here, that's what documentation should be like. If it's not clear enough from the documentation alone, then either your proc is too long (which I don't believe it is right now) or it's not clear enough in the way it works.

For instance, I'd name `locate_living` to something that's more indicative of what it is, i.e. `living_inside` or something like that.

Also, I'd probably do a check on the mind, because what if they get killed and then respawned using another  ghost role or something like that? I'm not the most familiar with ghost role code but that seems to be the sensible thing to do, to me.

Reply:
it does the ghost/mind check earlier on.